gradle-plugin/src/functionalTest/kotlin/dev/zacsweers/metro/gradle/incremental/MetroProject.kt

```diff
@@ -15,38 +16,40 @@ abstract class MetroProject(
 
   open fun StringBuilder.onBuildScript() {}
 
-  val gradleProject: GradleProject
+  open val gradleProject: GradleProject
     get() =
       newGradleProjectBuilder(DslKind.KOTLIN)
         .withRootProject {
           sources = this@MetroProject.sources()
-          withBuildScript {
-            plugins(GradlePlugins.Kotlin.jvm, GradlePlugins.metro)
+          withBuildScript { applyMetroDefault() }
+        }
+        .write()
 
-            withKotlin(
-              buildString {
-                onBuildScript()
+  fun BuildScript.Builder.applyMetroDefault() = apply {
+    plugins(GradlePlugins.Kotlin.jvm, GradlePlugins.metro)
 
-                // Metro config
-                appendLine("metro {")
-                appendLine(
-                  """
-                  debug.set(true)
-                  reportsDestination.set(layout.buildDirectory.dir("metro"))
-                """
-                    .trimIndent()
-                )
-                val metroOptions =
-                  metroOptions.enableScopedInjectClassHints
-                    ?.let { "enableScopedInjectClassHints.set($it)" }
-                    .orEmpty()
-                if (metroOptions.isNotBlank()) {
-                  appendLine("  $metroOptions")
-                }
-                appendLine("}")
-              }
-            )
-          }
+    withKotlin(
+      buildString {
+        onBuildScript()
+
+        // Metro config
+        appendLine("metro {")
+        appendLine(
+          """
+            debug.set($debug)
```

Comment:
Propagate debug value from the `MetroProject` constructor, it wasn't used before